feat: add bottom app launcher with configurable mode toggle

AyeZeeBB opened the pull request:
- Add BottomAppLauncher as alternative to side launcher
- Add useBottomLauncher config to switch between side/bottom modes
- Add bottomLauncherWidth setting (550-1200px)
- Update IPC toggleApplauncherMenu to respect launcher preference
- Simplify category filter with text labels instead of icons

AyeZeeBB committed:
feat(launcher): add bottom app launcher with configurable mode toggle

- Add BottomAppLauncher as alternative to side launcher
- Add useBottomLauncher config to switch between side/bottom modes
- Add bottomLauncherWidth setting (550-1200px)
- Update IPC toggleApplauncherMenu to respect launcher preference
- Simplify category filter with text labels instead of icons

AyeZeeBB committed:
Update BottomAppLauncher.qml

AyeZeeBB committed:
fix: respect useBottomLauncher setting for app button

- Add TOGGLE_BOTTOM_LAUNCHER event to EventNames
- Update LeftBar.updateGlobalState to emit event when bottom launcher enabled
- Add event listener in shell.qml to toggle bottom launcher on event

AyeZeeBB committed:
feat: Add command system with wallpaper selector to app launchers

Introduce a `>` command prefix system to both bottom and sidebar app launchers, enabling quick access to shell actions and utilities.

New components:
- CommandsRegistry.qml: Singleton for shared command definitions
- CommandItem.qml: Reusable command list item with consistent styling
- WallpaperSelector.qml: Full-featured wallpaper picker with three sources

Wallpaper selector features:
- Local wallpapers from built-in assets
- Custom folder selection via file dialog
- Wallhaven.cc API integration with:
  - Search functionality
  - Category filters (General/Anime/People)
  - Sorting options (Top/Latest/Random/Views)
  - Infinite scroll pagination
  - Automatic download to cache before applying.

AyeZeeBB committed:
fix: replace Custom folder with Downloaded source

- Rename "Custom" source tab to "Downloaded"
- Auto-refresh Downloaded list after successful Wallhaven download

AhmedSaadi0 commented on config/quickshell/components/WallpaperSelector.qml:
```diff
@@ -0,0 +1,1297 @@
+
+    function downloadWallhaven(wallpaperData) {
+        const filename = wallpaperData.id + "." + wallpaperData.file_type.split('/')[1];
+        const cachePath = App.cacheFolderPath + "/wallpapers";
```
I think it would be better if you created a variable in `App` for `downloadedWallpapers`.

AhmedSaadi0 commented on config/quickshell/components/WallpaperSelector.qml:
```diff
@@ -0,0 +1,1297 @@
+        wallhavenDownloadProcess.command = [
+            "sh", "-c",
+            "mkdir -p '" + cachePath + "' && curl -s -L -o '" + filePath + "' '" + wallpaperData.path + "'"
+        ];
```
The command also must be in `App.scripts.bash` with the other commands.

AhmedSaadi0 commented on config/quickshell/components/WallpaperSelector.qml:
```diff
@@ -0,0 +1,1297 @@
+    }
+
+    Process {
+        id: localWallpapersProcess
```
This process must be inside `WallpaperController.qml`, and there must be a variable like `property var wallpapersList: []` that holds the `cachedWallpapersList`.

And in this file you just read the data, so all the logic for wallpapers be only in one place as a single source of truth.

AhmedSaadi0 commented on config/quickshell/components/WallpaperSelector.qml:
```diff
@@ -0,0 +1,1297 @@
+    }
+
+    Process {
+        id: wallhavenProcess
```
Also this process must be in `WallpaperController.qml` and have 3 `signals ` `fetchingWallhavenWallpapersStarted`, `wallhavenWallpapersFetched(response)`, and `wallhavenWallpapersError(errorDetails)`

and in this file you just watch the signals.

AhmedSaadi0 commented on config/quickshell/components/WallpaperSelector.qml:
```diff
@@ -0,0 +1,1297 @@
+    }
+
+    Process {
+        id: wallhavenDownloadProcess
```
Also this must be moved to `WallpaperController.qml` and when wallpaper is downloaded succefully you just `push` `cachedWallpapersList`. and send a signal (optional)

AhmedSaadi0 commented on config/quickshell/components/WallpaperSelector.qml:
```diff
@@ -0,0 +1,1297 @@
+            }
+        }
+    }
+}
```
The file is so large, it needs to be separated into several files here `components/wallpaper_selector/**.qml`

AhmedSaadi0 reviewed (changes requested):
Everything is good, only the `WallpaperSelector.qml` needs some refactoring. If you're short on time or unable to address these changes right now, please let me know, and I'd be happy to take care of it.

AyeZeeBB committed:
refactor: centralize logic and split WallpaperSelector

- Add downloadedWallpapersPath and downloadWallpaperCommand to App.qml
- Move all wallpaper processes to WallpaperController.qml:
  - Local wallpapers process
  - Downloaded wallpapers process
  - Wallhaven search process
  - Wallhaven download process
- Add signals for async operations (fetchingStarted, fetched, error)

Split WallpaperSelector.qml into sub-components:
- WallhavenFilters.qml - sorting, categories, colors, resolution dropdowns
- WallpaperGrid.qml - GridView with loading/empty states
- WallpaperCard.qml - individual wallpaper item delegate
